iTunesSearch/iTunesSearch/ViewControllers/SearchResultsTableViewController.swift

```diff
@@ -0,0 +1,105 @@
+        let resultType: ResultType!
+        switch searchTypeSegmentedControl.selectedSegmentIndex {
+        case 0:
+            resultType = ResultType.software
+        case 1:
+            resultType = ResultType.musicTrack
+        default:
+            resultType = ResultType.movie
+        }
+        let limit = Int(resultsLimitSlider.value)
+        searchResultController.performSearch(searchTerm: searchTerm, resultType: resultType, limit: limit) { error in
+            if let error = error {
+                print("Search error: \(error)")
+            } else {
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                }
+            }
+        }
+    }
+    
+    @IBAction func resultsLimitSlider(_ sender: UISlider) {
+        resultsLimitLabel.text = "# of Results: \(Int(sender.value))"
+    }
+    
+    // MARK: - Table view data source
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchResultController.searchResult.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "searchResultCell", for: indexPath)
+        let searchResult = searchResultController.searchResult[indexPath.row]
+        cell.textLabel?.text = searchResult.title
+        cell.detailTextLabel?.text = searchResult.creator
+        return cell
+    }
+
+    // MARK: - Navigation
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "detailShowSegue" {
+            if let detailVC = segue.destination as? DetailViewController,
+                let indexPath = tableView.indexPathForSelectedRow {
+                detailVC.searchResultController = searchResultController
+                detailVC.searchResult = searchResultController.searchResult[indexPath.row]
+            }
+        }
+    }
+    
+}
+
+extension SearchResultsTableViewController: UISearchBarDelegate {
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let searchTerm = searchBar.text else { return }
+        searchBar.resignFirstResponder()
+        let resultType: ResultType!
+        switch searchTypeSegmentedControl.selectedSegmentIndex {
+        case 0:
+            resultType = ResultType.software
+        case 1:
+            resultType = ResultType.musicTrack
+        default:
+            resultType = ResultType.movie
+        }
+        let limit = Int(resultsLimitSlider.value)
+        searchResultController.performSearch(searchTerm: searchTerm, resultType: resultType, limit: limit) { error in
+            if let error = error {
+                print("Search error: \(error)")
+            } else {
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                }
+            }
+        }
```

Comment:
... down here. Maybe you could refactor this out into a function and call it from both places.